app/scripts/migrations/029.js

```diff
@@ -0,0 +1,26 @@
+// next version number
+const version = 29
+const failTxsThat = require('./fail-tx')
+
+// time
+const seconds = 1000
+const minutes = 60 * seconds
+const hours = 60 * minutes
+const unacceptableDelay = 12 * hours
+
+/*
+
+normalizes txParams on unconfirmed txs
+
+*/
+
+module.exports = {
+  version,
+
+  migrate: failTxsThat(version, 'Stuck in approved state for too long.', (txMeta) => {
+    const createdTime = txMeta.submittedTime
```

Comment:
Suppose a tx gets created with a gas price that is so low it will never be mined. Are there any negative consequences, e.g. related to nonces, to us flagging it as failed locally even though it is stilling pending on-chain?

Reply:
I think maybe this should only fail txs that are in the approved state? Should the return be something like `txMeta.status === "approved" && now - createdTime > unacceptableDelay`?

Otherwise, transactions of other statuses could be set to failed. I think this will address my previous concern as well.

Reply:
Ya that makes sense to me, we should filter based on the status

Reply:
Oh I completely agree, I missed that, thank you!